script.js

```diff
@@ -1,4 +1,152 @@
+//Compare combined numbers accross all players and higher combined number wins
+// Add score to winner
+// Press submit button to roll dice (set gameState to gameStateRollDice)
+
+var gameStateRollDice = "gamestate is roll dice";
+var gameStateChooseDice = "game state is choose dice";
+var gameStateCompareScore = "game state is comparing Scores";
+var gameState = gameStateRollDice;
+
+var currentPlayerDiceNumber = [];
+
+var currentPlayer = 1;
+var scoreBoard = [];
+
+var rollDice = function () {
+  // Generate a decimal from 0 through 6, inclusive of 0 and exclusive of 6.
+  var randomDecimal = Math.random() * 6;
+  // Remove the decimal with the floor operation.
+  // This will be an integer from 0 to 5 inclusive.
+  // Add 1 to get valid dice rolls of 1 through 6 inclusive.
+  var diceNumber = Math.floor(randomDecimal) + 1;
+  console.log("dicenumber = " + diceNumber);
+  return diceNumber;
+};
+
+// rolls 2 dice for each player
+var rollDiceForPlayer = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerDiceNumber.push(rollDice()); //using roll to generate numbers, using push() to add the numbers to currentPlayerDiceNumber string created
+    counter += 1;
+  }
+  console.log("Current player dice numbers are: " + currentPlayerDiceNumber);
+  return (
+    "Player " +
+    currentPlayer +
+    ",<br>Dice 1 = " +
+    currentPlayerDiceNumber[0] +
+    "<br>Dice 2 = " +
+    currentPlayerDiceNumber[1] +
+    "<br> Please input 1 (Dice1) or 2 (Dice2) to pick the dice result as the first digit"
+  );
+};
+
+//input validation, combine score, and scoreboard
+var playerScoreCounter = function (playerInput) {
+  var playerScore = "";
+  if (playerInput != 1 && playerInput != 2) {
+    console.log("error input");
+    return (
+      "Please enter 1 or 2 only. <br> Your dice rolls are: " +
+      currentPlayerDiceNumber[0] +
+      " & " +
+      currentPlayerDiceNumber[1]
+    );
+  }
+  if (playerInput == 1) {
+    console.log("input 1");
+    var playerScore = Number(
+      String(currentPlayerDiceNumber[0]) + String(currentPlayerDiceNumber[1])
+    );
+    scoreBoard.push(playerScore);
+    currentPlayerDiceNumber = [];
+    console.log(scoreBoard);
+    console.log(currentPlayerDiceNumber);
+    console.log("currentplayer1score : " + playerScore);
+    return "Your score is: " + playerScore;
+  }
+  if (playerInput == 2) {
+    console.log("input 2");
+    var playerScore = Number(
+      String(currentPlayerDiceNumber[1]) + String(currentPlayerDiceNumber[0])
+    );
+    scoreBoard.push(playerScore);
+    currentPlayerDiceNumber = [];
+    console.log(scoreBoard);
+    console.log(currentPlayerDiceNumber);
+    console.log("currentplayer2score: " + playerScore);
+    return "Your score is: " + playerScore;
+  }
+};
+
+var comparePlayerScore = function () {
+  var compareMessage =
+    "Player 1 score: " + scoreBoard[0] + "<br>Player 2 score: " + scoreBoard[1];
```

Comment:
Can consider initiating variables like this:
```
var player1Score = scoreBoard[0];
var player2Score = scoreBoard[1];
```

and then use these variables later on in this function instead of directly `scoreBoard[0/1]` just because it's not as intuitive what it might be referring to.